Apps/Database/src/Delegates/IDelegationDelegate.cs

```diff
@@ -36,19 +38,47 @@ public interface IDelegationDelegate
         Task<Dependent?> GetDependentAsync(string hdid, bool includeAllowedDelegation = false);
 
         /// <summary>
-        /// Fetches the dependent audits by hdid from the database.
+        /// Fetches the agent audit(s) by query options from the database.
         /// </summary>
-        /// <param name="hdid">The dependent hdid to query on.</param>
-        /// <returns>A list of Dependent Audits.</returns>
-        Task<IEnumerable<DependentAudit>> GetDependentAuditsAsync(string hdid);
+        /// <param name="query">The query criteria.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        Task<AgentAuditQueryResult> GetAgentAuditsAsync(AgentAuditQuery query);
 
         /// <summary>
-        /// Updates the dependent object including allowed delegation associations as well as resource delegates in the DB.
+        /// Adds or updates the dependent object including allowed delegation associations as well as resource delegates and agent
+        /// audit to the DB.
         /// </summary>
         /// <param name="dependent">The dependent to update.</param>
         /// <param name="resourceDelegatesToRemove">The resource delegates to remove.</param>
-        /// <param name="dependentAudit">The dependent audit to create.</param>
+        /// <param name="agentAudit">The agent audit to create.</param>
         /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
-        Task UpdateDelegationAsync(Dependent dependent, IEnumerable<ResourceDelegate> resourceDelegatesToRemove, DependentAudit dependentAudit);
+        Task UpdateDelegationAsync(Dependent dependent, IEnumerable<ResourceDelegate> resourceDelegatesToRemove, AgentAudit agentAudit);
+    }
+
+    /// <summary>
+    /// Agent audit query options to determine the agent audit results to return.
+    /// </summary>
+    public record AgentAuditQuery
+    {
+        /// <summary>
+        /// Gets the audit group to search by.
+        /// </summary>
+        public AuditGroup? GroupCode { get; init; }
+
+        /// <summary>
+        /// Gets the hdid to search by.
+        /// </summary>
+        public string Hdid { get; init; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Returns agent audit query results.
+    /// </summary>
+    public record AgentAuditQueryResult
```

Comment:
This type of wrapper really isn't adding benefits, we wrap in delegate then immediately unwrap in services, just to put it in a new wrapper for delegation info. its just adding complexity.

Reply:
This is following the pattern that @ytqsl introduced in DbResourceDelegateDelegate.  Using a 'Query' and a 'Result' object was to ensure that any future changes to this query would not affect an existing service call.  Comments? @MikeLyttle  @rayking29  @sslaws  @ytqsl